plutus-core/plutus-core/src/PlutusCore/Parser.hs

```diff
@@ -78,8 +77,8 @@ term = choice $ map try
 
 -- | Parse a PLC program. The resulting program will have fresh names. The underlying monad must be capable
 -- of handling any parse errors.
-parseProgram ::
-    ByteString -> Either (ParseErrorBundle T.Text ParserError) (Program TyName Name DefaultUni DefaultFun SourcePos)
+parseProgram :: (AsParserError e, MonadError e m) =>
```

Comment:
I'm confused: where's the `MonadQuote`? The parsers run in `MonadQuote` already, the idea was just to remove the `runQuote` in `parse`.